src/service/routing/core.clj

```diff
@@ -1,16 +1,43 @@
 (ns service.routing.core
-  (:require [compojure.api.sweet :refer :all]
-            [service.routing.spec]))
+  (:require [clojure.spec.alpha :as s]
+            [clojure.spec.gen.alpha :as gen]
+            [ring.util.http-response :refer [ok]]
+            [compojure.api.sweet :refer [context GET defapi]]
+            [service.routing.spec :as spec]
+            [service.routing.directions :as dir]
+            [service.routing.graph.generators :as g]))
+            ;[expound.alpha :as expound]
 
-(def app
-  (api
-    {:swagger
-     {:ui "/"
-      :spec "/swagger.json"
-      :data {:info {:title "Routing API"
-                    :description "Routing for hippos"}
-             :tags [{:name "direction", :description "direction similar to mabbox"}]}}}
+;(s/valid? (s/and ::radiuses-regex #(= (count coordinates) (count %)))))
+;(s/and ::s/radiuses-regex #(= (count coordinates) (count %))))
 
-    service.routing.spec/routes))
+;;;; TEST @mehdi
+;(expound/expound-str ::direction (dir/direction (gen/generate (g/graph 1000)) :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]))
+;(s/explain ::direction (dir/direction (gen/generate (g/graph 1000)) :coordinates [{:lon 1 :lat 2} {:lon 3 :lat 4}]))
+
+(defapi app
+  {:swagger {:ui "/"
+             :spec "/swagger.json"
+             :data {:info {:title "Routing API"
+                           :description "Routing for hippos"}
+                    :tags [{:name "direction", :description "direction similar to mabbox"}]}}}
+  (GET "spec/direction/:coordinates" []
+    :coercion :spec
+    :summary "direction with clojure.spec"
+    :path-params [coordinates :- ::spec/coordinates-raw]
+    :query-params [{steps :- boolean? false}
+                   {radiuses :- ::spec/radiuses-raw nil}
+                   {alternatives :- boolean? false}
+                   {language :- string? "en"}]
+    :return ::spec/direction
+    (ok (if-not (s/valid? (s/and #(= (count coordinates) (count %))) radiuses)
```

Comment:
@mehdisadeghi I made a mistake here. It only makes sense to compare the amount of elements if the radiouses are defined at all !

Sorry for introducing bugs in your code :/

Reply:
No problem at all! I'm looking at it now :)